tools/federation/src/main/java/org/eclipse/rdf4j/federated/FederationManager.java

```diff
@@ -93,10 +92,27 @@ public void init(FedX federation, FederationContext federationContext) {
 		this.federationContext = federationContext;
 		this.executor = Executors.newCachedThreadPool(new NamingThreadFactory("FedX Executor"));
 
-		updateStrategy();
+		updateFederationType();
 		reset();
 	}
 
+	/**
+	 * 
+	 * @return the initialized and configured {@link FederationEvaluationStrategyFactory}
+	 */
+	public synchronized FederationEvaluationStrategyFactory getFederationEvaluationStrategyFactory() {
```

Comment:
Not sure about the synchronized here the `strategyFactory` is not volatile and may be accessed without synchronized in the set method below.

Reply:
You are right. Not required here. I have slightly modified the code and also made this method package private